Add information on manually installing Typescript plugin. (#2727) (#2728)
withastro/docs: src/content/docs/en/guides/typescript.mdx

/// <reference types="astro/client" />
```

> If you are not using VSCode, it is also advisable to [install the typescript plugin from npm](https://www.npmjs.com/package/@astrojs/ts-plugin):

Reviewer: ```suggestion
If you are not using VSCode, you can install the [Astro TypeScript plugin]((https://www.npmjs.com/package/@astrojs/ts-plugin) to support importing `.astro` files from `.ts` files (which can be useful for re-exporting). 
```

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npm install @astrojs/ts-plugin
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm install @astrojs/ts-plugin
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn add @astrojs/ts-plugin
  ```
  </Fragment>
</PackageManagerTabs>

> Once you have done this, add the following to your `tsconfig.json`:

Reviewer: ```suggestion
Then, add the following to your `tsconfig.json`:
```

```json title="tsconfig.json"
  "compilerOptions": {
    "plugins": [
      {
        "name": "@astrojs/ts-plugin"
      },
    ],

>     "jsx": "react",

Reviewer: ```suggestion

```

Author: This is needed even when not using react in sublime text, can't speak for other editors

Reviewer: What if you're using SolidJS and would then need `jsx: preserve`? If it's necessary to have a JSX entry here, let's explain why and what values.

Reviewer 2: Any `jsx` setting should be fine. `preserve` is the safest one, since it'll work no matter which framework you're using.

```

> The plugin should now be working. To check, create a ts file and import an astro component into it. You should have no warning messages from your editor saying the import cannot be found.

Reviewer: ```suggestion
To check that the plugin is working, create a `.ts` file and import an Astro component into it. You should have no warning messages from your editor.
```

/// <reference types="astro/client" />
```

If you are not using VSCode, you can install the [Astro TypeScript plugin](https://www.npmjs.com/package/@astrojs/ts-plugin) to support importing `.astro` files from `.ts` files (which can be useful for re-exporting). 

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npm install @astrojs/ts-plugin
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm install @astrojs/ts-plugin
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn add @astrojs/ts-plugin
  ```
  </Fragment>
</PackageManagerTabs>

Then, add the following to your `tsconfig.json`:

```json title="tsconfig.json"
  "compilerOptions": {
    "plugins": [
      {
        "name": "@astrojs/ts-plugin"
      },
    ],

>     "jsx": "react",

Reviewer: ```suggestion
    "jsx": "preserve",
```
Switch to `preserve` as it's the safest option

```

To check that the plugin is working, create a `.ts` file and import an Astro component into it. You should have no warning messages from your editor.

> :::note
> The `jsx` setting specifies what JSX code is generated. You may have to change it depending on the particular framework you are using. For example, SolidJS users may prefer the "preserve" option. For plain Astro projects, "react" works well. Valid values are "preserve", "react", "react-jsx", "react-jsdev" and "react-native".
> :::

Reviewer: ```suggestion
```
I don't think this section is necessary. Most of these values don't help in Astro, and `jsx: preserve` always works.
